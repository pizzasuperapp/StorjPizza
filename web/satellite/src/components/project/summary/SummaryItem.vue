// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div class="summary-item" :style="{ backgroundColor: backgroundColor }">
        <p class="summary-item__title" :style="{ color: titleColor }">{{ title }}</p>
        <p v-if="isMoney" class="summary-item__value" :style="{ color: valueColor }">{{ value | centsToDollars }}</p>
        <p v-else class="summary-item__value" :style="{ color: valueColor }">{{ value }}</p>
    </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

// @vue/component
@Component
export default class SummaryItem extends Vue {
    @Prop({ default: '#fff' })
    public readonly backgroundColor: string;
    @Prop({ default: '#1b2533' })
    public readonly titleColor: string;
    @Prop({ default: '#000' })
    public readonly valueColor: string;
    @Prop({ default: '' })
    public readonly title: string;
    @Prop({ default: 0 })
    public readonly value: number;
    @Prop({ default: false })
    public readonly isMoney: boolean;
}
</script>

<style scoped lang="scss">
    .summary-item {
        padding: 25px 30px;
        width: 100%;
        border-radius: 6px;

        &__title {
            font-family: 'font_medium', sans-serif;
            font-size: 14px;
            line-height: 14px;
            margin: 0 0 12px;
        }

        &__value {
            font-family: 'font_bold', sans-serif;
            font-size: 28px;
            line-height: 23px;
            letter-spacing: -1px;
            margin: 0;
        }
    }
</style>
