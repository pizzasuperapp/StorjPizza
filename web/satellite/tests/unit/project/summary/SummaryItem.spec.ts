// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

import SummaryItem from '@/components/project/summary/SummaryItem.vue';

import { createLocalVue, mount } from '@vue/test-utils';

const localVue = createLocalVue();

localVue.filter('centsToDollars', (cents: number): string => {
    return `$${(cents / 100).toFixed(2)}`;
});

describe('SummaryItem.vue', (): void => {
    it('renders correctly if not money', (): void => {
        const wrapper = mount(SummaryItem, {
            localVue,
            propsData: {
                backgroundColor: '#fff',
                titleColor: '#1b2533',
                valueColor: '#000',
                title: 'test',
                value: 100,
                isMoney: false,
            },
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly if money', (): void => {
        const wrapper = mount(SummaryItem, {
            localVue,
            propsData: {
                backgroundColor: '#fff',
                titleColor: '#1b2533',
                valueColor: '#000',
                title: 'test',
                value: 100,
                isMoney: true,
            },
        });

        expect(wrapper).toMatchSnapshot();
    });
});
