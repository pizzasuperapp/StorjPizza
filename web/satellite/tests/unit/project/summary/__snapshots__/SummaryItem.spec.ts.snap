// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryItem.vue renders correctly if money 1`] = `
<div class="summary-item" style="background-color: rgb(255, 255, 255);">
  <p class="summary-item__title" style="color: rgb(27, 37, 51);">test</p>
  <p class="summary-item__value" style="color: rgb(0, 0, 0);">$1.00</p>
</div>
`;

exports[`SummaryItem.vue renders correctly if not money 1`] = `
<div class="summary-item" style="background-color: rgb(255, 255, 255);">
  <p class="summary-item__title" style="color: rgb(27, 37, 51);">test</p>
  <p class="summary-item__value" style="color: rgb(0, 0, 0);">100</p>
</div>
`;
